### PR TITLE
Patch to log the number of records in a transactional block

### PR DESCRIPTION
**Pull request by vaibhav-yb**
This PR aims to add a patch which is to log the number of records coming out in a transaction block between the `BEGIN` and `COMMIT` change records. A warning will be logged if the record count is 0.

### CODE REVIEW

File: debezium-connector-yugabytedb2/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java [hunk 1]
```diff
@@ -378,13 +399,24 @@ private void getChanges2(ChangeEventSourceContext context,
...
                                         LOGGER.debug("LSN in case of COMMIT is " + lsn);
                                         offsetContext.updateWalPosition(tabletId, lsn, lastCompletelyProcessedLsn, message.getCommitTime(),
                                                 String.valueOf(message.getTransactionId()), null, null/* taskContext.getSlotXmin(connection) */);
                                         commitMessage(part, offsetContext, lsn);
                                         dispatcher.dispatchTransactionCommittedEvent(part, offsetContext);
+
+                                        if (recordsInTransactionalBlock.get(tabletId) == 0) {
+                                            LOGGER.warn("Records in the transactional block for tablet {} are 0", tabletId);
+                                        } else {
+                                            LOGGER.info("Records in the transactional block for tablet {}: {}", tabletId, recordsInTransactionalBlock.get(tabletId));
```
Comment (vaibhav-yb) [inline debezium-connector-yugabytedb2/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java:415]: Make this a `LOGGER.debug()` statement.

File: debezium-connector-yugabytedb2/pom.xml [hunk 2]
```diff
@@ -41,12 +41,12 @@
...
             <groupId>org.yb</groupId>
             <artifactId>yb-client</artifactId>
-            <version>0.8.17-SNAPSHOT</version>
+            <version>0.8.15-SNAPSHOT</version>
         </dependency>
         <dependency>
             <groupId>org.yb</groupId>
             <artifactId>yb-client</artifactId>
-            <version>0.8.17-SNAPSHOT</version>
+            <version>0.8.15-SNAPSHOT</version>
```
Comment (vaibhav-yb) [inline debezium-connector-yugabytedb2/pom.xml:49]: Using this version number, create a release for `2.12.4.1-b2` and another release for the latest master of `yugabyte-db`

File: debezium-connector-yugabytedb2/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java [hunk 3]
```diff
@@ -364,12 +374,23 @@ private void getChanges2(ChangeEventSourceContext context,
...
                                         if (message.getOperation() == Operation.COMMIT) {
                                             LOGGER.debug("LSN in case of COMMIT is " + lsn);
                                             offsetContext.updateWalPosition(tabletId, lsn, lastCompletelyProcessedLsn, message.getCommitTime(),
                                                     String.valueOf(message.getTransactionId()), null, null/* taskContext.getSlotXmin(connection) */);
                                             commitMessage(part, offsetContext, lsn);
+
+                                            if (recordsInTransactionalBlock.get(tabletId) == 0) {
+                                                LOGGER.warn("Records in the transactional block for tablet {} are 0", tabletId);
+                                            } else {
+                                                LOGGER.info("Records in the transactional block for tablet {}: {}", tabletId, recordsInTransactionalBlock.get(tabletId));
```
Comment (suranjan) [inline debezium-connector-yugabytedb2/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java:389]: At info level there will be too much logs. Most of the time there will be single record transaction.

**Review by suranjan — CHANGES_REQUESTED** [review]
Ideally we can use just one map and contains method to see if the begin record has come.
In begin:
recordsInTransactionalBlock.put(tabletId, 0)
in dml record
if( recordsInTransactionalBlock.contains(tabletId)
recordsInTransactionalBlock.merge(tabletId, 1, Integer::sum)
//warning if it doesn't contain ..ideally that shouldn't be the case?

File: debezium-connector-yugabytedb2/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java [hunk 4]
```diff
@@ -249,7 +250,7 @@ private void bootstrapTablet(YBTable table, String tabletId) throws Exception {
         long term = getCheckpointResponse.getTerm();
         long index = getCheckpointResponse.getIndex();
         LOGGER.info("Checkpoint for tablet {} before going to bootstrap: {}.{}", tabletId, term, index);
-        if (term == 0 && index == 0) {
+        if (term == -1 && index == -1) {
```
Comment (suranjan) [inline debezium-connector-yugabytedb2/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java:253]: We don't have the change for -1,-1 OpId in final-connector?

**Review by suranjan — APPROVED** [review]
Looks good.

**Pull request merged**